conda/common/serialize/json.py

```diff
@@ -32,18 +35,83 @@ def default(self, obj: Any) -> Any:
         return super().default(obj)
 
 
-def dump(*args, **kwargs):
-    kwargs.setdefault("cls", CondaJSONEncoder)
-    kwargs.setdefault("indent", 2)
-    return json.dump(*args, **kwargs)
+@overload
+def write(obj: Any, *, fp: IO[str], **kwargs) -> None: ...
+
+
+@overload
+def write(obj: Any, *, path: PathType, **kwargs) -> None: ...
+
+
+@overload
+def write(obj: Any, **kwargs) -> str: ...
 
 
-def dumps(*args, **kwargs):
+def write(
+    obj: Any, *, fp: IO[str] | None = None, path: PathType | None = None, **kwargs
+) -> None | str:
+    if sum(value is not None for value in (fp, path)) > 1:
```

Comment:
Would this be clearer if we flip it?


```suggestion
    if (fp, path).count(None) < 1:
```

Reply:
hmmm, I think the most clear way to express this is to be more explicit. Like:
```
if fp and path:
  raise ValueError("At most one of fp or path must be provided")
```

If part of the motivation for this logic is to be able to easily extend this error checking to support more output types, then I think ` sum(value is not None for value in (fp, path)) > 1` is more correct.

For example consider the 3 values `a="a"`, `b="b"`, `c=None`. For providing all these values, we would want the error case to be triggered. Since we only want at most one of these values to be provided.
* ` sum(value is not None for value in (a,b,c)) >1` -> `True` error case is triggered
* `(a,b,c).count(None) < 1` -> `False` error is not triggered

So, comparing these solutions I think:
1. `if sum(value is not None for value in (fp, path)) ` is good, though maybe too future looking
2. `if fp and path` if the simplist and more explicit implementation
3. `if (fp, path).count(None) < 1` is not more clear than (1) imo, I think it's my least preferred option.

Would you be satisfied sticking with (1)? Open to other options too!